climakitae/data_loaders.py

```diff
@@ -205,108 +250,111 @@ def _process_and_concat(selections, location, dsets, cat_subset):
+                            ssp_filename = fnmatch.filter(
+                                list(dsets.keys()),
+                                "*{0}*{1}*{2}*".format(
+                                    activity_id,
+                                    simulation,
+                                    _scenario_to_experiment_id(scenario),
+                                ),
+                            )[0]
+                    except:  # Some simulation + ssp options are not available. Just continue with the loop if no filename is found
+                        continue
+                    # Grab data
+                    historical_data = dsets[historical_filename]
+                    ssp_data = dsets[ssp_filename]
+
+                    # Concatenate data. Rename scenario attribute
+                    # This will append the SSP data to the historical data, both with the same simulation
+                    ds_sim = xr.concat(
+                        [historical_data, ssp_data],
+                        dim="time",
+                        coords="minimal",
+                        compat="override",
+                        join="inner",
+                    )
 
-            else:
-                try:
-                    if (  # Need to get CESM2 data if ensmean is selected for ssp2-4.5 or ssp5-8.5
-                        simulation == "ensmean"
-                    ) and (
-                        scenario
-                        in [
-                            "SSP 2-4.5 -- Middle of the Road",
-                            "SSP 5-8.5 -- Burn it All",
-                        ]
-                    ):
-                        filename = [
-                            name
-                            for name in dsets.keys()
-                            if "CESM2." + _scenario_to_experiment_id(scenario) in name
-                        ][0]
-                    else:
-                        filename = [
-                            name
-                            for name in dsets.keys()
-                            if simulation + "." + _scenario_to_experiment_id(scenario)
-                            in name
-                        ][0]
-                except:
-                    continue
-                sim_list.append(dsets[filename][selections.variable_id])
+                else:
+                    try:
+                        if (  # Need to get CESM2 data if ensmean is selected for ssp2-4.5 or ssp5-8.5
+                            simulation == "ensmean"
+                        ) and (
+                            scenario
+                            in [
+                                "SSP 2-4.5 -- Middle of the Road",
+                                "SSP 5-8.5 -- Burn it All",
+                            ]
+                        ):
+                            filename = fnmatch.filter(
+                                list(dsets.keys()),
+                                "*{0}*CESM2*{1}*".format(
```

Comment:
I don't love that this is hard-coded at 'CESM2' -- we'd like this to just work if, in principle, another simulation for one of those SSPs was ever added to the catalog.

Reply:
This is unrelated to the LOCA data actually. It's only used when retrieving ensemble means (see the code directly above it) for SSP 245 and SSP 585.  I don't think our existing TMY code ever uses those two scenarios, and neither scenario has an ensemble mean... so I'm not sure why this code I added this code in ages ago. The TMY stuff has undergone so many iterations, I will need to check in with Victoria on it's current state. 

It's showing up here as a diff because I changed the way the filenames are found for all cases. I used to use list comprehension but prefer the wildcard matching that fnmatch has, so I switched it for all cases in this code chunk.

Reply:
I'll make a trello card to look into this

Reply:
https://trello.com/c/ijwDPxxG/373-cesm2-data-retrieval-for-ssp-245-and-585